File: src/03.Aplas_B1/dummy/app/src/main/java/org/aplas/basicappx/MainActivity.java
package org.aplas.basicappx;

import androidx.appcompat.app.AppCompatActivity;

import androidx.appcompat.app.AlertDialog;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {
    private Distance dist;
    private Weight weight;
    private Temperature temp;
    private Button convertBtn;
    private EditText inputTxt;
    private EditText outputTxt;
    private Spinner unitOri;
    private Spinner unitConv;
    private RadioGroup unitType;
    private CheckBox roundBox;
    private ImageView imgView;
    private AlertDialog startDialog;
    private CheckBox formBox;


    MainActivity() {
        this.dist = new Distance();
        this.weight = new Weight();
        this.temp = new Temperature();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        convertBtn = (Button) findViewById(R.id.convertButton);
        inputTxt = (EditText) findViewById(R.id.inputText);
        outputTxt = (EditText) findViewById(R.id.outputText);
        unitOri = (Spinner) findViewById(R.id.oriList);
        unitConv = (Spinner) findViewById(R.id.convList);
        unitType = (RadioGroup) findViewById(R.id.radioGroup);
        roundBox = (CheckBox) findViewById(R.id.chkRounded);
        formBox = (CheckBox) findViewById(R.id.chkFormula);
        imgView = (ImageView) findViewById(R.id.img);

        unitType.setOnCheckedChangeListener(
                new RadioGroup.OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(RadioGroup group, int checkedId) {
                        RadioButton resrab = (RadioButton) findViewById(checkedId);
                        ArrayAdapter<CharSequence> arr;
                        inputTxt.setText("0");
                        outputTxt.setText("0");
                        if (resrab.getText().equals("Temperature")) {
                            arr = ArrayAdapter.createFromResource(unitType.getContext(), R.array.tempList, android.R.layout.simple_spinner_item);
                            imgView.setImageResource(R.drawable.temperature);
                            imgView.setTag(R.drawable.temperature);
                            arr.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
                            unitOri.setAdapter(arr);
                            unitConv.setAdapter(arr);
                        } else if (resrab.getText().equals("Distance")) {
                            arr = ArrayAdapter.createFromResource(unitType.getContext(), R.array.distList, android.R.layout.simple_spinner_item);
                            imgView.setImageResource(R.drawable.distance);
                            imgView.setTag(R.drawable.distance);
                            arr.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
                            unitOri.setAdapter(arr);
                            unitConv.setAdapter(arr);
                        } else {
                            arr = ArrayAdapter.createFromResource(unitType.getContext(), R.array.weightList, android.R.layout.simple_spinner_item);
                            imgView.setImageResource(R.drawable.weight);
                            imgView.setTag(R.drawable.weight);
                            arr.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
                            unitOri.setAdapter(arr);
                            unitConv.setAdapter(arr);
                        }

                    }
                });

        convertBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                doConvert();
            }
        });

        unitOri.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                doConvert();
            }
            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                return;
            }
        });

        unitConv.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                doConvert();
            }
            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                return;
            }
        });

        roundBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                doConvert();
            }
        });

        formBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                ((ImageView)findViewById(R.id.imgFormula)).setVisibility((isChecked)?View.VISIBLE:View.INVISIBLE);(isChecked) ? imgView.setVisibility(imgView.VISIBLE) : imgView.setVisibility(imgView.INVISIBLE) ;
            }
        });
    }


    @Override
    protected void onStart() {
        super.onStart();

        startDialog = new AlertDialog.Builder(MainActivity.this).create();
        startDialog.setTitle("Application started");
        startDialog.setMessage("This app can use to convert any units");
        startDialog.setButton(AlertDialog.BUTTON_NEUTRAL, "OK",
                new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });
        startDialog.show();
    }



    protected double convertUnit(String sa, String sb, String sc, double da) {

        if (sa.equalsIgnoreCase("Temperature")) {
            return temp.convert(sb, sc, da);
        } else if (sa.equalsIgnoreCase("Distance")) {
            return dist.convert(sb, sc, da);
        } else {
            return weight.convert(sb, sc, da);
        }
    }

    protected String strResult(double da, boolean ba) {
        DecimalFormat dc = new DecimalFormat("#.#####");
        DecimalFormat dc2 = new DecimalFormat("#.##");
        if (ba) {
            return dc2.format(da);
        } else {
            return dc.format(da);
        }
    }


    protected void doConvert() {
        double in = Double.parseDouble(inputTxt.getText().toString());
        RadioButton rab = (RadioButton) findViewById(unitType.getCheckedRadioButtonId());
        double res = convertUnit(rab.getText().toString(), unitOri.getSelectedItem().toString(), unitConv.getSelectedItem().toString(), in);
        outputTxt.setText(strResult(res, roundBox.isChecked()));
    }



}

File: src/05.Kotlin/resources compability/app/src/main/java/android/example/com/MainActivity.kt
package android.example.com

import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity



class MainActivity : AppCompatActivity() {

    private lateinit var diceImage: ImageView
    private lateinit var diceImage2: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.roll_button)

        diceImage = findViewById(R.id.dice_image)
        diceImage2 = findViewById(R.id.dice_image2)
        rollButton.setOnClickListener { rollDice() }
//        add_button.setOnClickListener { add1() }
    }

    private fun rollDice() {
        Toast.makeText(this, "rolled", Toast.LENGTH_SHORT).show()
        val randomInt = (1..6).random()
        val randomInt2 = (1..6).random()

        val drawableResource = when (randomInt) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }

        val drawableResource2 = when (randomInt2) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }

        diceImage.setImageResource(drawableResource)
        diceImage2.setImageResource(drawableResource2)
    }

//    private fun add1(){
//        val textView: TextView = findViewById<TextView>(R.id.result_text)
//        var res = textView.text.toString()
//        if(res.equals("Dice Rolled !")){
//            Toast.makeText(this, "value initialized", Toast.LENGTH_SHORT).show()
//            textView.text = "1";
//        }else if(res.equals("6")){
//            Toast.makeText(this, "value is max", Toast.LENGTH_SHORT).show()
//            textView.text = "6"
//        }else{
//            Toast.makeText(this, "added 1", Toast.LENGTH_SHORT).show()
//            res = (res.toInt()+1).toString()
//            textView.text = res
//        }
//    }
}